apps/src/dance/Dance.js

```diff
@@ -122,17 +122,19 @@ Dance.prototype.init = function (config) {
     this.studioApp_.init(config);
     this.currentCode = this.studioApp_.getCode();
     if (this.usesPreview) {
+      // rerender preview each time student code changes
       this.studioApp_.addChangeHandler(e => {
-        // We want to check if the workspace code changed only when a block has been moved or
-        // if a block has changed.
-        // A move event is fired when a block is dragged and then dropped.
         if (
-          e.type !== Blockly.Events.BLOCK_MOVE &&
+          e.type !== Blockly.Events.BLOCK_DRAG &&
           e.type !== Blockly.Events.BLOCK_CHANGE
         ) {
           return;
         }
 
+        if (e.type === Blockly.Events.BLOCK_DRAG && e.isStart) {
```

Comment:
Unrelated optimization to only rerender the preview when a drag ends, rather than when it starts and ends.

Reply:
Thanks for doing this!